workflow/validate/validate.go

```diff
@@ -172,7 +175,13 @@ func ValidateWorkflow(wftmplGetter templateresolution.WorkflowTemplateNamespaced
 	}
 
 	if !opts.IgnoreEntrypoint {
-		_, err = ctx.validateTemplateHolder(&wfv1.WorkflowStep{Template: wf.Spec.Entrypoint}, tmplCtx, &wf.Spec.Arguments, map[string]interface{}{})
+		var args wfv1.ArgumentsProvider
+		args = &wf.Spec.Arguments
```

Comment:
Why not 
```suggestion
		args := &wf.Spec.Arguments
```
Instead of `var args wfv1.ArgumentsProvider`?

Reply:
FakeArgument is a separate struct.  `ArgumentsProvider` is a common interface that used to pass either wfv1. Argument or FakeArgument.

Reply:
Ah, of course!